### ISSUE TITLE
Provisioning Framework - issue with Directory Handler

### ISSUE THREAD

**Issue by luismanez** [OfficeDev/PnP-Sites-Core#756]
#### Category
[X ] Bug
[ ] Enhancement

#### Environment
[ X] Office 365 / SharePoint Online
[ ] SharePoint 2016
[ ] SharePoint 2013

#### Expected or Desired Behavior
You can specify a Directory node in the PnP Schema with no "MetadataMappingFile" attribute (is optional), and the Handler shouldn't raise an error.

#### Observed Behavior
When the MetadataMappingFile attribute is not set, code raises an error, as is trying to load a file with empty name. This is because the GetMetadataProperties method is just checking if the value is null, but not Empty:

```c#
if (directory.MetadataMappingFile != null)
```

And when the attribute is loaded from the XML, and is not present in the XML, is set as Empty.

#### Steps to Reproduce
Add a Directory node without specifying the MetadataMappingFile attribute

I'm about to submit a Pull Request to fix that issue.

**Comment by luismanez** [comment]
Also, if the IncludedExtensions attribute is not set (is also optional), the Schema XML is setting a default value to "*.*", and that makes an issue when the IncludedExtensions attribute is processed later, to get the relevant file from the Directory, as the current code won't get any File because this code:

```c#
if (!String.IsNullOrEmpty(directory.IncludedExtensions))
            {
                var includedExtensions = directory.IncludedExtensions.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
                files = files.Where(f => includedExtensions.Contains($"*{Path.GetExtension(f).ToLower()}")).ToList();
            }
```
The Where filter never matches, when the extension is *.*

**Comment by VesaJuvonen** [comment]
Fixed with #757 by Luis.
